### PR TITLE
frame counter; reduce unnecessary processing

### PR DESCRIPTION
**Pull request by dalekube**
Instead of reading, resizing, converting, and appending every single frame in the video, this code maintains a counter and frame step interval to selectively process only frames that are necessary in (total_frames/frame_step/2) intervals. This reduces the overall processing time and splits the frame selection in relatively equal intervals across the video.

### CODE REVIEW

**Comment by metal3d** [comment]
Hello,
Yes! That's something that I wanted to make later, but you did it :)

Can you add a CONTRIBUTORS file with your name inside? Push that in that PR - I will make some checks with my notebook (later there will be nose tests) and merge this. Then I will make new release.

Thanks a lot !

**Comment by dalekube** [comment]
Happy to help! Thanks for letting me contribute.

File: src/keras_video/generator.py [hunk 1]
```diff
@@ -191,38 +192,38 @@ def __getitem__(self, index):
 
             if video not in self.__frame_cache:
                 cap = cv.VideoCapture(video)
+                total_frames = cap.get(cv2.CAP_PROP_FRAME_COUNT)
```
[the remark this answers is not in the document]
Comment (metal3d) [reply src/keras_video/generator.py]: Made in #12 PR wich contains your commits, your own will be closed.
Please take a look on the new PR, I also made changes for SlidingFrameGenerator.

Thanks a lot !

**Comment by metal3d** [comment]
Made in #12 PR wich contains your commits, your own will be closed.
Please take a look on the new PR, I also made changes for SlidingFrameGenerator.
I've fixed the `cv2` usage also.

Thanks a lot !

**Pull request closed without merging**